spec/linter-stylelint-spec.js

```diff
@@ -263,4 +265,27 @@ describe('The stylelint provider for Linter', () => {
       expect(messages.length).toBe(0);
     });
   });
+
+  describe('will autofix what errors it can', () => {
+    it('saves fixed css if attemptFix is true', async () => {
+      atom.config.set('linter-stylelint.attemptFix', true);
+      atom.config.set('linter-stylelint.disableWhenNoConfig', false);
+      const fixedEditor = await atom.workspace.open(doneFixCSS);
```

Comment:
Correct, the editor being linted needs to be a real file, I was just saying that the expected text doesn't need to be and could be inlined.

----

Linter should never call `lint()` on an unsaved (no path) editor, however it does happen sometimes so there _should_ be a guard in there preventing this from going anywhere, but it doesn't look like there is in this provider. Not that that would change your spec there 😉.